client/src/utils/lmcSimulator.js

```
@@ -265,44 +309,38 @@ class LMC {
-              };
-              this.documentations.push(docObj);
-              throw new Error("Input required"); // Custom error to signal input request
+              this.halted = true;
+              this.error = "Input required.";
             } else {
               throw new Error("Input buffer is empty.");
             }
           }
-          this.accumulator = this.inputs.shift();
+          const input = parseInt(this.inputs.shift(), 10);
+          if (isNaN(input)) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)